src/scanelf.d

```diff
@@ -42,152 +42,178 @@ void scanElfObjModule(void delegate(const(char)[] name, int pickAny) pAddSymbol,
     {
         printf("scanElfObjModule(%s)\n", module_name);
     }
-    const buf = base.ptr;
-    const buflen = base.length;
-    int reason = 0;
-    if (buflen < Elf32_Ehdr.sizeof)
+
+    void corrupt(int reason)
     {
-        reason = __LINE__;
-    Lcorrupt:
         error(loc, "corrupt ELF object module %s %d", module_name, reason);
-        return;
     }
-    if (memcmp(buf, elf, 4))
-    {
-        reason = __LINE__;
-        goto Lcorrupt;
-    }
-    if (buf[EI_VERSION] != EV_CURRENT)
+
+    if (base.length < Elf32_Ehdr.sizeof)
+        return corrupt(__LINE__);
+
+    immutable ubyte[4] elf = [0x7F, 'E', 'L', 'F']; // ELF file signature
```

Comment:
make it `static` lest you end up initializing it every pass through the function